packages/core/src/modules/mdoc/Mdoc.ts

```diff
@@ -75,23 +75,33 @@ export class Mdoc {
     return this.issuerSignedDocument.issuerSigned.issuerAuth.decodedPayload.validityInfo
   }
 
-  public get deviceSignedNamespaces(): MdocNameSpaces {
+  public get deviceSignedNamespaces(): MdocNameSpacesRecord {
     if (this.issuerSignedDocument instanceof DeviceSignedDocument === false) {
       throw new MdocError(`Cannot get 'device-namespaces from a IssuerSignedDocument. Must be a DeviceSignedDocument.`)
     }
 
-    return this.issuerSignedDocument.allDeviceSignedNamespaces
+    return Object.fromEntries(
+      Array.from(this.issuerSignedDocument.allDeviceSignedNamespaces.entries()).map(([namespace, value]) => [
+        namespace,
+        Object.fromEntries(Array.from(value.entries())),
+      ])
```

Comment:
Yes, see https://github.com/animo/mdoc/pull/39.

The library was encoding the namespaces as an object rather than a map, which resulted in the Google mdoc library not being able to verify the signature as the signature input data on their side was different.

The mdoc library only transformed it to a map on our side just for the signature part, but still included the wrong data inside the device signed. I updated our mdoc implementation to now always use a map. For credo it's still useful/simple to use a record.

Will also open an issue in the auth0 mdl lib